File: bms/go/geo/geo_r2_Rect_InteriorContains/geo_r2_Rect_InteriorContains.go
package test

var ()

const ()

func (r geo_r2_Rect) InteriorContains(other geo_r2_Rect) bool {
	return r.X.InteriorContainsInterval(other.X) && r.Y.InteriorContainsInterval(other.Y)
}

func (i geo_r1_Interval) InteriorContainsInterval(oi geo_r1_Interval) bool {
	if oi.IsEmpty() {
		return true
	}
	return i.Lo < oi.Lo && oi.Hi < i.Hi
}

func (i geo_r1_Interval) IsEmpty() bool { return i.Lo > i.Hi }

type geo_r2_Rect struct {
	X, Y geo_r1_Interval
}

type geo_r1_Interval struct {
	Lo, Hi float64
}

File: translations_only/success/go/geo/geo_r2_Rect_InteriorContains/gpt4/Restart/geo_r2_Rect_InteriorContains.rs

struct GeoR2Rect {
    x: GeoR1Interval,
    y: GeoR1Interval,
}

struct GeoR1Interval {
    lo: f64,
    hi: f64,
}

impl GeoR2Rect {
    fn interior_contains(&self, other: &GeoR2Rect) -> bool {
        interior_contains_interval(&self.x, &other.x) && interior_contains_interval(&self.y, &other.y)
    }
}

fn interior_contains_interval(i: &GeoR1Interval, oi: &GeoR1Interval) -> bool {
    if is_empty(oi) {
        return true;
    }
    i.lo < oi.lo && oi.hi < i.hi
}

fn is_empty(i: &GeoR1Interval) -> bool {
    i.lo > i.hi
}

fn main() {}
